tests/functional/detection_status/test_detection_status.py

```diff
@@ -100,9 +100,21 @@ def _create_source_file(self, version):
         with open(os.path.join(self._test_dir, self._source_file), 'w') as f:
             f.write(self.sources[version])
 
-        codechecker.check(self._codechecker_cfg,
+        codechecker.analyze(self._codechecker_cfg,
+                            'hello',
+                            self._test_dir)
+
+        try:
```

Comment:
Couldn't we create a separate test for the metadata.json removal? At least a separate test case if not a separate test package.